### PR TITLE
Split analysis steps out of dev/bots/test.dart into dev/bots/analyze.dart

### PR DESCRIPTION
**Pull request by aam**
This allows to run analysis step with command line arguments that are only applicable to flutter analyze(like --dart-sdk, needed for dart-flutter-engine head-head-head bot).

### CODE REVIEW

**Comment by athomas** [comment]
@aam did you forget to add analyze.dart to the commit?

**Comment by aam** [comment]
> @aam did you forget to add analyze.dart to the commit?

Oops, I certainly did!

File: dev/bots/test.dart [hunk 1]
```diff
@@ -30,7 +29,7 @@ const String arrow = '⏩';
 const String clock = '🕐';
 
 const Map<String, ShardRunner> _kShards = <String, ShardRunner>{
-  'analyze': _analyzeRepo,
+  // 'analyze': _analyzeRepo, handled by analyze.dart
```
Comment (gspencergoog) [inline dev/bots/test.dart]: Just remove this line, and the ones below.  You might add a comment above the declaration just saying that this only covers shards handled by this script.

File: dev/bots/analyze.dart [hunk 2]
```diff
@@ -0,0 +1,605 @@
...
+  }
+
+  return result;
+}
+
+String elapsedTime(DateTime start) {
+  return new DateTime.now().difference(start).toString();
+}
+
+Future<Null> _runCommand(String executable, List<String> arguments, {
```
Comment (gspencergoog) [inline dev/bots/analyze.dart]: There's a lot of repeated code here... Can you please split out the common code and import it into both from there?  Most of the command execution and progress stuff is identical.

**Review by gspencergoog — APPROVED** [review]

**Comment by gspencergoog** [comment]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Pull request merged**